### ISSUE TITLE
HexChat is not DPI-aware

### ISSUE THREAD

**Issue by StepS-** [hexchat/hexchat#828]
HexChat is not DPI-aware. Especially a major issue on Windows 8.1 with a DPI scaling setting other than 96/120 DPI (other than 100% or 125%).

Can be solved at the user's side by enabling "Disable DPI scaling on high screen resolutions" in the "Compatibility" tab but is not a long-term solution.

**Comment by realies** [comment]
Would be greatly appreciated if DPI issues are fixed across Windows and OS X.
There's also no way of fixing the pixelated tray icon in Windows 10 with 150% scaling, adding a custom one in the config folder makes no difference as the software presumably forces it to render as 16x16.

<img src="http://i.snag.gy/3a5MB.jpg">

**Comment by TingPing** [comment]
So this was brought up in #hexchat again and I think it is worth considering disable the scaling on Windows still. With the default theme most of Gtk uses Windows to draw things and they come out scaled properly so only a portion of the widgets are not scaled and look small:

- https://i.imgur.com/C2ZaV1A.png  
- https://i.imgur.com/GiUIWhS.png  
- https://i.imgur.com/6V3OYNB.png  
- https://i.imgur.com/NFwOCUZ.png  

The question is just which is more important, crisp text in a text heavy application or proper scaling for a few icons here and there. Though it may lead to other unexpected issues.

**Comment by nwgat** [comment]
i stopped using hexchat because of this, FYI
https://msdn.microsoft.com/en-us/library/windows/desktop/dd371771

the GTk or hexchat devs need to fix this, its been like this for years now
this should have been fixed 3 years ago, why? more and more people get higher resolution (4k) displays these days and the first thing they will notice is the blurrymess that is hexchat

**Comment by TingPing** [comment]
@nwgat The last major release of Gtk2 was in 2011 and it is now in maintenance mode.

**Comment by mdmower** [comment]
@TingPing Re: "The question is just which is more important, crisp text in a text heavy application or proper scaling for a few icons here and there. Though it may lead to other unexpected issues."

I'm fully in support of crisp text as the highest priority. This is IRC we're talking about. Those who use IRC regularly use it for discussion (i.e. the _text_) much more so than for a pretty interface.

**Comment by xaxxon** [comment]
I just installed hexchat and now I'm looking for an alternative.   The text is just too blurry to deal with, especially when switching back and forth between something like chrome with crisp text.. it hurts every time I switch apps.

**Comment by tomek** [comment]
@xaxxon did you disable DPI scaling for HexChat? On 4K screen with 150% scaling - HC looks fine

**Comment by mB-PiBox** [comment]
@xaxxon right-click hexchat shortcut and then Properties. Under Compatibility, set Disable Display Scaling On High DPI... Its not perfect, but its functional. I have to do this on my Surface Pro 4.

**Comment by xaxxon** [comment]
sweet.  good call.

**Comment by adomas187** [comment]
Wow this "Disable Display Scaling on High DPi..." really worked very well. Also for few other apps it solved the problem. That's great!

**Comment by andreyrd** [comment]
FWIW, looks like this setting is not available with the Windows Store version, so I went back to the regular version.

**Comment by thomasfedb** [comment]
I'm having a few issues with widget sizing on a 4k display under Linux (Fedora). Also it seems like the chat windows gutter width (for timestamp and nick) doesn't scale either.

**Comment by jmtd** [comment]
This bug is pretty old and most of the discussion seems to be about the experience on Windows, but since various other DPI issues have been closed as duplicates of this one, I guess this is the right place for all DPI/scaling problems across all platforms…

The label "upstream" was applied 5 years ago because of lack of DPI reporting support in GTK+ in Windows specifically. I don't know whether that was addressed for Windows in the last 5 years but it has been for at least Linux I believe. Someone with more Windows or GTK+ on Windows experience may be able to tell whether that label still applies. I'm a little worried having a catch-all DPI issue labelled "upstream" means devs will pass over the issue even though there might still be work to do in Hexchat itself.

The current behaviour I experience wrt DPI and hexchat is, if I change my DPI/scaling settings on Linux, I have to restart hexchat, but it picks up my preferences at start up. It just does not react to changes at runtime. Does this match other (Linux users) experiences?

**Comment by TingPing** [comment]
>  I don't know whether that was addressed for Windows in the last 5 years but it has been for at least Linux I believe.

Yes upstream Gtk3 now supports hidpi. I will change the label to better reflect it is just HexChat that needs ported to Gtk > 2.

**Comment by EwoutH** [comment]
This is the current HexChat App from the Windows Store on Windows 10 1809 with 150% scaling. Only using one monitor, looks the same after restart.

<img alt="screenshot_156" src="https://user-images.githubusercontent.com/15776622/51831281-043c4700-22f2-11e9-825a-dbde7ab80c91.png">

**Comment by katanakiwi** [comment]
> The current behaviour I experience wrt DPI and hexchat is, if I change my DPI/scaling settings on Linux, I have to restart hexchat, but it picks up my preferences at start up. It just does not react to changes at runtime. Does this match other (Linux users) experiences?

This matches my experience. Using Mint 19 and i3wm 4.14.

**Comment by TreeBranches** [comment]
HexChat on my Surface Book 2 using the Windows Store version looks like garbage. This issue is _years_ old now. 😕 I can't justify the donation price of £8 when issues like this remain unresolved for so long, sorry.

I'll keep track of this issue for the supposed 
![Screenshot (6)](https://user-images.githubusercontent.com/8653544/57816146-2909dc80-7772-11e9-833e-be6ac16991dc.png)
time this _does_ get fixed, because overall I do prefer the HexChat interface/client features, but until then, I've had to stick to mIRC

**Comment by EwoutH** [comment]
Microsoft has an excellent guide on how to make applications DPI aware on Windows
- [High DPI Desktop Application Development on Windows](https://docs.microsoft.com/en-us/windows/desktop/hidpi/high-dpi-desktop-application-development-on-windows#dpi-awareness-mode)

**Comment by canton7** [comment]
They changed where the compatibility settings are on Windows 10.

Shortcut - Properties - Change high DPI settings - Check "Override high DPI scaling behavior", and set "Scaling performed by" to "Application".

**Comment by TreeBranches** [comment]
> Microsoft has an excellent guide on how to make applications DPI aware on Windows
> 
> * [High DPI Desktop Application Development on Windows](https://docs.microsoft.com/en-us/windows/desktop/hidpi/high-dpi-desktop-application-development-on-windows#dpi-awareness-mode)

Microsoft also have an excellent guide on how to update applications to be HighDPI aware.

This issue should 100% not be dumped on the user - it's an application problem.

https://docs.microsoft.com/en-us/windows/win32/hidpi/high-dpi-desktop-application-development-on-windows

**Comment by jmtd** [comment]
The exact same link was posted two comments earlier.

**Comment by samirsharma-github** [comment]
This works for Windows 20H2:
![image](https://user-images.githubusercontent.com/76894659/111029883-28513880-8425-11eb-8543-6c0be1085b45.png)

**Comment by orcus-de** [comment]
It looks like linuxmint will drop hexchat as its preinstalled irc client with upcoming mint 21.1 due to the lacking hidpi support (which will be a great loss for the irc support).

In case someone is working on hidpi support, I would dearly love to forward a related note to the devs.

cheers orcus
